src/app/reporting/Engine.cpp

```diff
@@ -336,6 +377,62 @@ void Engine::Run()
         mCurReadHandlerIdx = (mCurReadHandlerIdx + 1) % CHIP_IM_MAX_NUM_READ_HANDLER;
         readHandler        = imEngine->mReadHandlers + mCurReadHandlerIdx;
     }
+    for (auto & dirtyPath : mDirtyPathList)
+    {
+        dirtyPath.ClearDirty();
+    }
+    return;
+}
+
+void Engine::SetDirty(ClusterInfo & aClusterInfo)
+{
+    InteractionModelEngine * imEngine = InteractionModelEngine::GetInstance();
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+    ScopedLock lock(*this);
+#endif // !CHIP_SYSTEM_CONFIG_NO_LOCKING
+
+    for (int i = 0; i < CHIP_IM_MAX_NUM_READ_HANDLER; ++i)
+    {
+        ReadHandler * readHandler = &imEngine->mReadHandlers[i];
+
+        if (readHandler->IsSubscription() && readHandler->IsReportable())
+        {
+            ClusterInfo * clusterInstance = readHandler->GetAttributeClusterInfolist();
+
+            while (clusterInstance != nullptr)
+            {
+                bool setSuccess = false;
+                // If dirty path intersects with interested path from handler, it mark this in dirty path list and also
```

Comment:
For now, we need a additional check for the existence of the path.
For subscription, we should not report non-exist path.

Reply:
agree. would do it in seperate pr